sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatchRequestOptions.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.RequestOptions;
+import com.azure.cosmos.util.Beta;
+
+/**
+ * Encapsulates options that can be specified for a {@link TransactionalBatch}.
+ */
+@Beta(Beta.SinceVersion.V4_7_0)
+public final class TransactionalBatchRequestOptions {
+    private ConsistencyLevel consistencyLevel;
+    private String sessionToken;
+
+    /**
+     * Gets the consistency level required for the request.
+     *
+     * @return the consistency level.
+     */
+    public ConsistencyLevel getConsistencyLevel() {
+        return consistencyLevel;
+    }
+
+    /**
+     * Sets the consistency level required for the request.
+     *
+     * @param consistencyLevel the consistency level.
+     * @return the TransactionalBatchRequestOptions.
+     */
+    TransactionalBatchRequestOptions setConsistencyLevel(ConsistencyLevel consistencyLevel) {
```

Comment:
for CosmosItemRequestOption we didn't allow setting consistency level per request. To be consistent we shouldn't have that public method here either. if we decide to enable this feature we should consistently enable this everywhere.

Reply:
We do have consistencyLevel in CosmosItemRequestOption

Reply:
the ones in `CosmosItemRequestOption` are not public.
here `getConsistencyLevel` is public. the ask is to make it non-public.

Reply:
Oh!!! My apologies. Misunderstood it. Will make the change. Thanks!

Reply:
Done